Sources/PenguinGraphs/GraphTransformations.swift

```diff
@@ -415,3 +415,209 @@ extension BidirectionalGraph {
+  fileprivate var extraEdges: ExtraEdges
+
+  /// Creates a graph based on `base`, augmented with edges in `extraEdges`.
+  public init(_ base: Base, _ extraEdges: ExtraEdges) {
+    self.base = base
+    self.extraEdges = extraEdges
+  }
+
+  /// A collection of edges from a single vertex in `self`.
+  public typealias VertexEdgeCollection =
+    ConcatenatedCollection<
+      LazyMapCollection<Base.VertexEdgeCollection, EdgeId>,
+      LazyMapCollection<ExtraEdges.VertexEdgeCollection, EdgeId>>
+
+  /// A collection of edges from a single vertex.
+  public func edges(from vertex: VertexId) -> VertexEdgeCollection {
+    base.edges(from: vertex).lazy.map(EdgeId.a).concatenated(with:
+      extraEdges.edges(from: vertex).lazy.map(EdgeId.b))
+  }
+
+  /// Returns the source vertex of `edge`.
+  public func source(of edge: EdgeId) -> VertexId {
+    switch edge {
+    case .a(let edge): return base.source(of: edge)
+    case .b(let edge): return extraEdges.source(of: edge)
+    }
+  }
+
+  /// Returns the destionation vertex of `edge`.
+  public func destination(of edge: EdgeId) -> VertexId {
+    switch edge {
+    case .a(let edge): return base.destination(of: edge)
+    case .b(let edge): return extraEdges.destination(of: edge)
+    }
+  }
+}
+
+extension UnionEdgesGraph: BidirectionalGraph where Base: BidirectionalGraph, ExtraEdges: BidirectionalGraph {
+  /// A collection of edges to a vertex.
+  public typealias VertexInEdgeCollection = ConcatenatedCollection<
+    LazyMapCollection<Base.VertexInEdgeCollection, EdgeId>,
+    LazyMapCollection<ExtraEdges.VertexInEdgeCollection, EdgeId>>
+
+  /// Returns a collection of all edges in `self` whose destination is `vertex`.
+  public func edges(to vertex: VertexId) -> VertexInEdgeCollection {
+    base.edges(to: vertex).lazy.map(EdgeId.a).concatenated(with:
+      extraEdges.edges(to: vertex).lazy.map(EdgeId.b))
+  }
+}
+
+extension UnionEdgesGraph: PropertyGraph
+where Base: PropertyGraph, ExtraEdges: PropertyGraph, ExtraEdges.Edge == Base.Edge {
+  /// Arbitrary data asssociated with every vertex.
+  public typealias Vertex = Base.Vertex
+
+  /// Arbitrary data associated with every edge.
+  public typealias Edge = Base.Edge
+
+  /// Accesses the data associated with `vertex`.
+  public subscript(vertex vertex: VertexId) -> Vertex {
+    get { base[vertex: vertex] }
+    _modify { yield &base[vertex: vertex] }
+  }
+
+  /// Access the data associated with `edge`.
+  public subscript(edge edge: EdgeId) -> Edge {
+    get {
+      switch edge {
+      case .a(let edge): return base[edge: edge]
+      case .b(let edge): return extraEdges[edge: edge]
+      }
+    }
+
+    _modify {
+      switch edge {
+      case .a(let edge): yield &base[edge: edge]
+      case .b(let edge): yield &extraEdges[edge: edge]
+      }
+    }
+  }
+}
+
+extension UnionEdgesGraph: VertexListGraph where Base: VertexListGraph {
+  /// The number of vertices in `self`.
+  public var vertexCount: Int { base.vertexCount }
```

Comment:
I initially added it because I thought that in some graphs, it might be possible that `vertexCount` can be more efficient than `vertices.count`. That said, the more I think about it, I think that a proper graph implementation should make `vertices.count` be exactly the same efficiency as `vertexCount` and so it should probably be deprecated & eventually removed. (https://github.com/saeta/penguin/issues/88)